### ISSUE TITLE
Bad Accuracy when converting QKeras Model

### ISSUE THREAD

**Issue by LordScarface** [fastmachinelearning/hls4ml#397]
Hello again,

I'm trying to convert a QKeras (or after that AutoQKeras) Model using hls4ml. Now the unquantized Model converts just fine and the Accuracy of the converted HLS Model is the same as the input Keras Model. When I now try to convert the QKeras Model, Accuracy of the HLS Model is ~10% (vs. ~99% for the quantized QKeras Model). 

The Models I used for testing (MNIST Dataset):
[- Keras Model ~99% Accuracy when converted with hls4ml](https://github.com/fastmachinelearning/hls4ml/files/7179641/keras_mnist.zip)
[- QKeras Model ~10% Accuracy when converted with hls4ml](https://github.com/fastmachinelearning/hls4ml/files/7179647/qkeras_mnist_quant.zip)

My Code for loading and evaluating the Models:
```python
from tensorflow.keras.datasets import mnist
from tensorflow.keras.layers import *
from tensorflow.keras.models import Model
from tensorflow.keras.optimizers import *
from tensorflow.keras.utils import to_categorical
import numpy as np

from qkeras.autoqkeras import *
from qkeras import *
from qkeras.utils import model_quantize
from qkeras.qtools import run_qtools
from qkeras.qtools import settings as qtools_settings

from sklearn.metrics import accuracy_score

import hls4ml

from qkeras.utils import _add_supported_quantized_objects

def print_dict(d, indent=0):
    align=20
    for key, value in d.items():
        print('  ' * indent + str(key), end='')
        if isinstance(value, dict):
            print()
            print_dict(value, indent+1)
        else:
            print(':' + ' ' * (20 - len(key) - 2 * indent) + str(value))

def get_dataset():
    (x_train, y_train), (x_test, y_test) = mnist.load_data()
    
    x_train = (x_train/255.0).astype(np.float32)
    x_test = (x_test/255.0).astype(np.float32)
    
    x_train = x_train.reshape(x_train.shape[0], 28,28,1)
    x_test = x_test.reshape(x_test.shape[0], 28,28,1)
        
    y_train = to_categorical(y_train, 10)
    y_test = to_categorical(y_test, 10)
    
    return (x_train, y_train), (x_test, y_test)

(x_train, y_train), (x_test, y_test) = get_dataset()

co = {}
_add_supported_quantized_objects(co)

# hls4ml_input_model = tf.keras.models.load_model('keras_mnist.h5', custom_objects=co)
hls4ml_input_model = tf.keras.models.load_model('qkeras_mnist_quant.h5', custom_objects=co)

cfg = hls4ml.utils.config_from_keras_model(hls4ml_input_model, granularity='name')

for layer in cfg['LayerName'].keys():
    cfg['LayerName'][layer]['Trace'] = True

print_dict(cfg)

hls_model = hls4ml.converters.convert_from_keras_model(hls4ml_input_model,
                                                       hls_config=cfg,
                                                       io_type='io_stream',
                                                       output_dir='mnist_qkeras', # or mnist_keras
                                                       part='xczu7ev-ffvc1156-2-e')
hls_model.compile()

from sklearn.metrics import accuracy_score

y_hls = hls_model.predict(np.ascontiguousarray(x_test))

print("hls4ml Accuracy: {}".format(accuracy_score(np.argmax(y_test, axis=1), np.argmax(y_hls, axis=1))))
```

**Comment by thesps** [comment]
Hi, have you tried using the profiling & tracing tools in hls4ml to pinpoint the first layer that has an issue? Perhaps you can include some of the plots which are generated. If you haven't used them before they are introduced in [the tutorial](https://github.com/fastmachinelearning/hls4ml-tutorial) part 2, with [documentation here](https://fastmachinelearning.org/hls4ml/api/profiling.html). The `compare` method in profiling may also be helpful ([documentation here](https://fastmachinelearning.org/hls4ml/autodoc/hls4ml.model.html#hls4ml.model.profiling.compare)).
